botocore/stub.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2014 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import copy
+from botocore.validate import validate_parameters
+from botocore.exceptions import ParamValidationError, StubResponseError
+from botocore.vendored.requests.models import Response
+
+
+class QueueStubber(object):
+    """
+    This class will allow you to stub out requests so you don't have to hit
+    an endpoint to write tests. Responses are returned first in, first out.
+    If operations are called out of order, or are called with no remaining
+    queued responses, an error will be raised.
+    """
+    def __init__(self, client):
+        """
+        :param client: The client to add your stubs to.
+        """
+        self.client = client
+        self._queue = []
+
+    def activate(self):
+        """
+        Activates the stubber on the client
+        """
+        self.client.meta.events.register(
```

Comment:
I would suggest using a `unique_id` here to ensure we don't register this same handler twice.  Otherwise if you call `activate()` twice you'll have identical handlers here.

This *shouldn't* matter because of the short circuit behavior of before-call but it's still something that could cause issues in the future.